### PR TITLE
Azhu/anno tree temp

### PR DESCRIPTION
**Pull request by ang-zhu**
Has getArticleAnnotations return array-tree of the annotations and their children, up to depth 7 (checked with Peter). For deeper trees, should probably run a separate query.
* Articles now only keep track of their top-level annotations

### CODE REVIEW

**Comment by ang-zhu** [comment]
I didn't update the tests, so they're still failing. I can look into that if we want?

**Comment by charblanc** [comment]
I do believe this is GOOD TO GO.

File: server/controllers/annotation_controller.js [hunk 1]
```diff
@@ -63,7 +64,7 @@ export const editAnnotation = (userId, annotationId, updateText) => {
 // Removes an annotation from the database, updates the parent's numChildren
 // and recurses if the parent needs to be removed as well.
 const deleteAnnotationHelper = (user, annotation) => {
-  if (annotation.parent === null) { // base case: annotation is top-level
+  if (annotation.parent == null) { // base case: annotation is top-level
```
Comment (ang-zhu) [inline server/controllers/annotation_controller.js:66]: Not much difference between === and ==, but I think for consistency we want to stick with the triple?

File: server/models/annotation.js [hunk 2]
```diff
@@ -43,13 +46,14 @@ annotationSchema.pre('save', function preSave(next) {
 
   const fillReply = new Promise((resolve, reject) => {
     if (this.parent) {
-      this.constructor.findByIdAndUpdate(this.parent, { $inc: { numChildren: 1 } })
+      this.constructor.findByIdAndUpdate(this.parent, { $inc: { numChildren: 1 }, $push: { childAnnotations: this._id } }) // @angela: here is where we add childAnnotations
```
Comment (ang-zhu) [inline server/models/annotation.js]: This looks good to me, haven't tested yet though -- did you?

File: server/models/annotation.js [hunk 3]
```diff
@@ -43,13 +46,14 @@ annotationSchema.pre('save', function preSave(next) {
...
       .then((parent) => {
         this.article = parent.article;
         this.articleText = parent.articleText;
         this.ranges = parent.ranges;
         this.isPublic = parent.isPublic;
         this.groups = parent.groups;
+        this.isTopLevel = false; // @angela: maybe we don't need to do this , see virtual?
```
Comment (ang-zhu) [inline server/models/annotation.js]: We can go either way with this, just need to decide.

File: server/controllers/annotation_controller.js (hunk 1 above)
Comment (charblanc) [reply server/controllers/annotation_controller.js:66]: Actually... huge difference lol. If an annotation doesn't have a parent then the "parent" field is undefined, and not null

File: server/models/annotation.js (hunk 3 above)
Comment (charblanc) [reply server/models/annotation.js]: I tried for fun and got rid of this, replacing it with virtual column -- should test

**Pull request merged**